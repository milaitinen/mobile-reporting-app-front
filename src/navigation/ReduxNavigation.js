//Redux is not currently used, but this is kept for possible future use
/*
import * as ReactNavigation from 'react-navigation';
import React from 'react';
import { AppNavigation } from 'react-native';
import { connect } from 'react-redux';

// here is our redux-aware our smart component
function ReduxNavigation (props) {
    const { dispatch, nav } = props;
    const navigation = ReactNavigation.addNavigationHelpers({
        dispatch,
        state: nav
    });

    return <AppNavigation navigation={navigation} />;
}

const mapStateToProps = state => ({ nav: state.nav });
export default connect(mapStateToProps)(ReduxNavigation);
*/
