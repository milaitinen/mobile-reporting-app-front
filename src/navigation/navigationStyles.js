import EStyleSheet from 'react-native-extended-stylesheet';
import { Platform } from 'react-native';

const navigationStyles = EStyleSheet.create({

    HeaderContainer: {
        backgroundColor: '$darkBlue',
        borderBottomWidth: 0,
        borderBottomColor: 'transparent',
        paddingTop: Platform.OS === 'ios' ? 15 : 0,
        flexDirection: 'row',
        elevation: 0,
        height: 56,
        '@media (max-width: 350)': {
            paddingHorizontal: 7,
        },
        '@media (min-width: 350)': {
            paddingHorizontal: 10,
        },
    },

    titleContainer: {
        flex:1,
        justifyContent: 'center',
        alignItems: 'center'
    },

    ScreenHeader: {
        color: '#fff',
        fontWeight: 'normal',
        fontFamily: '$primaryFont',
        textAlignVertical: 'center',
        textAlign: 'center',
        '@media (max-width: 350)': {
            fontSize: 25,
        },
        '@media (min-width: 350)': {
            fontSize: 30,
        },

    },

    headerBackStyle: {
        color: '#fff',
        '@media (max-width: 350)': {
            fontSize: 30,
        },
        '@media (min-width: 350)': {
            fontSize: 30,
        },

    },

    backButtonContainer: {
        paddingLeft: 10,
        marginTop: Platform.OS === 'ios' ? 12 : 0 ,
        top: 0,
        position: 'absolute',
    },

    menuIconContainer: {
        paddingRight: 0,
        paddingLeft: 25,
        marginVertical: 5,
        marginTop: Platform.OS === 'ios' ? 8 : 0 ,
        top: 12,
        position: 'absolute',
    },


    menuIcon: {
        color: '#fff',
        '@media (max-width: 350)': {
            fontSize: 32,
        },
        '@media (min-width: 350)': {
            fontSize: 32,
        },
    },

});

export default navigationStyles;