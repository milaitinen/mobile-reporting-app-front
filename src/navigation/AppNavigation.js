import React from 'react';
import Text from 'react-native';
import { StackNavigator, DrawerNavigator } from 'react-navigation';
import { Icon } from 'react-native-elements';

import LoginScreen from '../screens/LoginScreen';

import TemplateScreen from '../screens/TemplateScreen';
import NewReportScreen from '../screens/NewReportScreen';

import navigationStyles from './navigationStyles';

import Sidebar from '../navigation/Sidebar';

import { strings } from '../locales/i18n';


//The stack that is contained within the drawer stack
const TemplateStack = StackNavigator({
    Templates: {
        screen: TemplateScreen,
        navigationOptions: ({ navigation }) => ({
            flex: 0.3,
            headerTitle: strings('templates.templates') ,
            headerStyle: navigationStyles.HeaderContainer,
            headerTitleStyle: navigationStyles.ScreenHeader,
            headerLeft:
                <Icon
                    name={'menu'}
                    type={'feather'}
                    color={'#fff'}
                    size={35}
                    containerStyle={navigationStyles.menuIcon}
                    onPress={() => { navigation.navigate('DrawerOpen'); }}>
                </Icon>
        })
    },
    Reports: {
        screen: TemplateScreen,
        navigationOptions: ({ navigation }) => ({ title: navigation.state.routeName })
    },
    NewReport: {
        screen: NewReportScreen,
        navigationOptions: strings('createNew.createNew')
    },

}, {
    // is this part necessary?
});

const DrawerStack = DrawerNavigator({
    Menu: {
        screen: TemplateStack,
        navigationOptions: strings('templates.templates')
    },
}, {
    //This loads the contents of the drawer from the custom Sidebar
    contentComponent: Sidebar,

    //These fix a bug with the drawer navigator
    drawerOpenRoute: 'DrawerOpen',
    drawerCloseRoute: 'DrawerClose',
    drawerToggleRoute: 'DrawerToggle',
});

const LoginStack = StackNavigator({
    loginScreen: {
        screen: LoginScreen
    },

}, {
    headerMode: 'screen',
    navigationOptions: {
        headerStyle: { backgroundColor: '#f0f8ff' },
        header: null
    }
});

// Manifest of possible screens
const MainScreenNavigator = StackNavigator({
    loginStack: { screen: LoginStack },
    drawerStack: { screen: DrawerStack },
}, {
    // Default config for all screens
    headerMode: 'none',
    title: 'Main',
    initialRouteName: 'loginStack',
});

export default MainScreenNavigator;
