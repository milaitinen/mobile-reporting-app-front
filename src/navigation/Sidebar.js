import { NavigationActions } from 'react-navigation';
import React from 'react';
import { Button, View } from 'react-native';
import { strings } from '../locales/i18n';
import { insertToken, insertUsername } from '../redux/actions/user';
import { emptyTemplates } from '../redux/actions/templates';
import { emptyReports } from '../redux/actions/reports';
import { connect } from 'react-redux';
import { LOGGED_OUT_ROUTE_NAME } from './AppNavigation';

class Sidebar extends React.Component {

    signOut = () => {
        // perform other logging out related tasks here

        this.props.dispatch(insertUsername(null));
        this.props.dispatch(insertToken(null));
        this.props.dispatch(emptyTemplates());
        this.props.dispatch(emptyReports());

        /*
        This sets the navigation back to the beginning, i.e. to the login screen.
        This means that the back button in Android will not return to the signed in part of the app,
        but will instead exit the app.
         */
        const actionToDispatch = NavigationActions.reset({
            index: 0,
            key: null,
            actions: [NavigationActions.navigate({ routeName: LOGGED_OUT_ROUTE_NAME })]
        });
        this.props.navigation.dispatch(actionToDispatch);
    };

    // TODO: update the look and consider removing inline style
    render () {
        return (
            <View style={ { padding: 20 } }>
                <Button onPress={() => this.signOut()} title={strings('navigation.signOut')}/>
            </View>
        );
    }
}

export default connect()(Sidebar);