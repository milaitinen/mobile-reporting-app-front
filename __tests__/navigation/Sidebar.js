import 'react-native';
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import { Sidebar } from '../../src/navigation/Sidebar';

configure({ adapter: new Adapter() });

describe('Sidebar', () => {
    const dispatch = jest.fn();
    const navigation = { state: { params: { isEditable: true } }, dispatch: dispatch };
    const wrapper = renderer.create(<Sidebar navigation={navigation}/>);

    it('should render correctly', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should call signOut() when Sign out button pressed', () => {
        const spy = sinon.spy(Sidebar.prototype, 'signOut');
        const sidebar = shallow(<Sidebar dispatch={dispatch} navigation={navigation} />);
        sidebar.find('Button').simulate('press');
        expect(spy.called).toEqual(true);
    });

});