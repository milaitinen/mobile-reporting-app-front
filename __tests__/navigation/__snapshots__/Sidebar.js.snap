// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render correctly 1`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        navigation.signOut
      </Text>
    </View>
  </View>
</View>
`;
